node/node.go

```diff
@@ -351,6 +366,16 @@ func (n *Node) Run(ctx context.Context) {
 		return
 	}
 
+	if n.plugin != nil {
+		go func() {
+			<-ctx.Done()
+			err := n.plugin.Shutdown()
```

Comment:
I think we should wait while `Shutdown()` function completes, right now there is no waitgroup. Maybe it's better to create some wrapper-service that calls plugin `Shutdown` in `Run(ctx)` wdyt?

Reply:
Yup, sounds good to me. Just upgraded the plugin to a service.